src/Core/Settings.h

```diff
@@ -196,6 +196,7 @@ class IColumn;
     M(Bool, allow_experimental_inverted_index, false, "If it is set to true, allow to use experimental inverted index.", 0) \
     \
     M(UInt64, mysql_max_rows_to_insert, 65536, "The maximum number of rows in MySQL batch insertion of the MySQL storage engine", 0) \
+    M(Bool, output_format_mysql_types, false, "Use MySQL converted types when connected via MySQL compatibility", 0) \
```

Comment:
All LGTM except this setting name and description.

It does not force output MySQL types everywhere, only for SHOW COLUMNS. For example, it does not affect Native, JSON, and CSVWithNamesAndTypes output formats.

Let's rename it to `use_mysql_types_in_show_columns`?